src/main/scala/net/snowflake/spark/snowflake/io/StageWriter.scala

```diff
@@ -281,16 +281,19 @@ private[io] object StageWriter {
     val writeTableState = new WriteTableState(conn)
 
     try {
+      val tableExists = if (params.checkTableExistenceInCurrentSchemaOnly){
```

Comment:
Change all of these assignments to just call DefaultJDBCWrapper.tableExists, where tableExists checks the parameter and uses a new connection

Reply:
follow it.